features/device_key/source/DeviceKey.h

```diff
@@ -0,0 +1,142 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DEVICEKEY_H
+#define MBED_DEVICEKEY_H
+
+#include "stddef.h"
+#include "stdint.h"
+#include "platform/NonCopyable.h"
+
+#if (NVSTORE_ENABLED)
+
+namespace mbed {
+/** \addtogroup drivers */
+
+#define DEVICE_KEY_16BYTE 16
+#define DEVICE_KEY_32BYTE 32
+
+enum DeviceKeyStatus {
```

Comment:
@mazimkhan 
Regarding the error handling. mbed-os 5.9 will introduce a new MbedErrorStatus mechanism and we will move to it immediately  when it will come available. Till then we want from one hand to keep the look alike of enum with negative error number as errors today are normally negative number.